components/Main/Workouts/AddWorkout/AddWorkout.js

```diff
@@ -2,7 +2,26 @@ import { StyledAddWorkout, StyledAddWorkoutHeader } from "./StyledAddWorkout";
 import Link from "next/link";
 import Image from "next/image";
 
-const AddWorkout = () => {
+const AddWorkout = ({ onAddWorkout }) => {
+  async function handleSubmit(event) {
+    event.preventDefault();
+    event.stopPropagation();
+    const name = event.target.elements.name.value;
+    const newWorkout = {
+      name: name,
+    };
+
+    await fetch("/api/workouts", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(newWorkout),
+    });
+    onAddWorkout(newWorkout);
```

Comment:
ich glaube, ich würde die Logik hier anders machen. du rufst ja schon die Funktion onAddWorkout in Zeile 21 auf. Ich würde dieser Funktion die POST-Logik geben und dahinter noch ein fetch von den geupdateten Daten - damit aktualisiert sich auch direkt dein State, was du gerade in der onAddWorkout-Funktion machst.